project/BuildSettings.scala

```diff
@@ -132,14 +132,11 @@ object BuildSettings {
   /** Docker settings, used by BE */
   lazy val dataflowDockerSettings = Seq(
     Docker / maintainer := "Snowplow Analytics Ltd. <support@snowplowanalytics.com>",
-    dockerBaseImage := "snowplow/k8s-dataflow:0.2.0",
-    dockerEnvVars := Map("HOME" -> "/tmp", "JAVA_OPTS" -> "-Duser.home=$HOME"),
-    Docker / daemonUser := "snowplow",
+    dockerBaseImage := "adoptopenjdk:11-jre-hotspot-focal",
+    Docker / defaultLinuxInstallLocation := "/opt/snowplow",
+    Docker / daemonUser := "daemon",
+    dockerEnvVars := Map("JAVA_OPTS" -> "-Duser.home=$HOME"),
```

Comment:
I'm very interested in this line!  Which dependency needs `user.home` to be set?  is it a dataflow thing?  And which directory does it use by default if this system property is not set?

It would be nice if `JAVA_OPTS` is kept as an environment variable that the user is allowed to change at run time, e.g.

```
docker run -e "JAVA_OPTS=-Dsome.other.property=something" snowplow/beam-enrich
```

But that example would now overwrite the `-Duser.home=$HOME` setting.

Having said that..... I am completely OK if there is no better way of making this work.

Reply:
Scio [uses](https://github.com/spotify/scio/blob/0539a4678d702503919f234be20a604cef4b13dc/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L147) `user.home` java system property for an internal regex check.

When `user.home` is `?`, Scio fails with

```scala
Exception in thread "main" java.util.regex.PatternSyntaxException: Dangling meta character '?' near index 0  ?/\..+/.+
```

and `?` happens when there is no corresponding username for a given (UID, GID) within the container. That line was a dirty hack to make Scio think that `user.home` isn't `?`.

While writing this comment, I just noticed a simple mistake. The reason of this issue is not having [this configuration](https://github.com/snowplow/enrich/blob/master/project/BuildSettings.scala#L128) for beam enrich as well. Without `Docker / daemonUserUid := None`, beam-enrich isn't run with `daemon` but with `USER 1001`. Using `daemonUserUid ` config makes beam-enrich get executed by `daemon` user where `user.home` isn't `?`. I just pushed a new commit on top of this branch to be squashed after the review.

Reply:
Thank you for your interest @istreeter ! what looks like a small mistake (not having `daemonUserUid` config for beam-enrich) made me spend a good time diving into scio codebase to find the regex check followed by finding a workaround etc.